clamav/clamav.go

```diff
@@ -0,0 +1,95 @@
+package clamav
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/dutchcoders/go-clamd"
+	"gitlab.com/NebulousLabs/errors"
+)
+
+// ClamAV is a client that allows scanning of content for malware.
+type ClamAV struct {
+	staticClam   *clamd.Clamd
+	staticPortal string
+}
+
+// New creates a new ClamAV client that will try to connect to the ClamAV
+// service listening on a TCP socket at the given address and port. Before
+// returning the client, New verifies the connection to ClamAV.
+func New(clamIP, clamPort, portal string) (*ClamAV, error) {
+	var err error
+	defer func() {
+		if err1 := recover(); err1 != nil {
+			err2 := errors.New(fmt.Sprintf("error while trying to connect to ClamAV: %v", err1))
+			err = errors.Compose(err, err2)
+		}
+	}()
+	clam := &ClamAV{
+		staticClam:   clamd.NewClamd(fmt.Sprintf("tcp://%s:%s", clamIP, clamPort)),
+		staticPortal: portal,
```

Comment:
Follow up: need a check that `portal` isn't an empty string.

Reply:
should also check that clamIP and clamPort aren't blank too right?

Reply:
A check for `portal` - sure, makes sense.

We don't need to check the clam params - if those are not valid we won't be able to connect to clamav and then ping it, so we'll return an error anyway.